modular_skyrat/modules/medical/code/wounds/synth/robotic_blunt.dm

```diff
@@ -0,0 +1,1255 @@
+#define PERCUSSIVE_MAINTENANCE_WOUND_SCANNED_CHANCE_MULT 1.5
+/// If the attacker is NOT our victim, chance of percussive maintenance succeeding is multiplied against this.
+#define PERCUSSIVE_MAINTENANCE_ATTACKER_NOT_VICTIM_CHANCE_MULT 2.5
+
+/datum/wound/blunt/robotic/proc/victim_attacked(datum/source, damage, damagetype, def_zone, blocked, wound_bonus, bare_wound_bonus, sharpness, attack_direction, attacking_item)
+	SIGNAL_HANDLER
+
+	if (def_zone != limb.body_zone) // use this proc since receive damage can also be called for like, chems and shit
+		return
+	if(!victim)
+		return
+
+	var/effective_damage = (damage - blocked)
+
+	var/obj/item/stack/gauze = limb.current_gauze
+	if(gauze)
+		effective_damage *= gauze.splint_factor
+
+	switch (limb.body_zone)
+
+		if(BODY_ZONE_CHEST)
+			var/oscillation_mult = 1
+			if (victim.body_position == LYING_DOWN)
+				oscillation_mult *= OSCILLATION_ATTACKED_LYING_DOWN_EFFECT_MULT
+			var/oscillation_damage = effective_damage
+			var/stagger_damage = oscillation_damage * chest_attacked_stagger_mult
+			if (victim.has_status_effect(/datum/status_effect/determined))
+				oscillation_damage *= ROBOTIC_WOUND_DETERMINATION_STAGGER_MOVEMENT_MULT
+			if ((stagger_damage >= chest_attacked_stagger_minimum_score) && prob(oscillation_damage * chest_attacked_stagger_chance_ratio))
+				stagger(stagger_damage * oscillation_mult, attack_direction, attacking_item, shift = stagger_damage * stagger_shake_shift_ratio)
+
+	if(!uses_percussive_maintenance() || damage < percussive_maintenance_damage_min || damage > percussive_maintenance_damage_max || damagetype != BRUTE || sharpness)
+		return
+	var/success_chance_mult = 1
+	if (HAS_TRAIT(src, TRAIT_WOUND_SCANNED))
+		success_chance_mult *= PERCUSSIVE_MAINTENANCE_WOUND_SCANNED_CHANCE_MULT
+	var/mob/living/user
+	if (isatom(attacking_item))
+		var/atom/attacking_atom = attacking_item
+		user = attacking_atom.loc // nullable
+
+		if (istype(user))
+			if (HAS_TRAIT(user, TRAIT_DIAGNOSTIC_HUD))
+				success_chance_mult *= PERCUSSIVE_MAINTENANCE_DIAG_HUD_CHANCE_MULT
+
+		if (user != victim)
+			success_chance_mult *= PERCUSSIVE_MAINTENANCE_ATTACKER_NOT_VICTIM_CHANCE_MULT // encourages people to get other people to beat the shit out of their limbs
+	if (prob(percussive_maintenance_repair_chance * success_chance_mult))
+		handle_percussive_maintenance_success(attacking_item, user)
+	else
+		handle_percussive_maintenance_failure(attacking_item, user)
+
+#undef OSCILLATION_ATTACKED_LYING_DOWN_EFFECT_MULT
+#undef PERCUSSIVE_MAINTENANCE_DIAG_HUD_CHANCE_MULT
+#undef PERCUSSIVE_MAINTENANCE_WOUND_SCANNED_CHANCE_MULT
+#undef PERCUSSIVE_MAINTENANCE_ATTACKER_NOT_VICTIM_CHANCE_MULT
+
+/// The percent, in decimal, of a stagger's shake() duration, that will be used in a addtimer() to queue aftershock().
+#define STAGGER_PERCENT_OF_SHAKE_DURATION_TO_AFTERSHOCK_DELAY 0.65 // 1 = happens at the end, .5 = happens halfway through
```

Comment:
This is something that's frequently happening in this file, please leave a whitespace between defines and procs, so it's easy to see the difference between procs at a glance.

You also forgot to document the proc itself, and it seems like most of the new procs in this file haven't been documented either.
```suggestion
#define STAGGER_PERCENT_OF_SHAKE_DURATION_TO_AFTERSHOCK_DELAY 0.65 // 1 = happens at the end, .5 = happens halfway through

```